bitnami/geode/templates/haproxy-configmap.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.metrics.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ printf "%s-haproxy-conf" (include "common.names.fullname" .) }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: metrics
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+data:
+  haproxy.cfg: |
+    global
+      log stdout format raw local0
+      maxconn 1024
+    defaults
+      log global
+      timeout client 60s
+      timeout connect 60s
+      timeout server 60s
+    frontend fe_main
+      bind :{{ .Values.metrics.containerPort }}
+      default_backend be_main
+    backend be_main
+      server web1 localhost:{{ .Values.locator.containerPorts.metrics }} check
```

Comment:
I see that the Metrics service is exposed only to `localhost` as it is hardcoded in the code, so the proxy is actually required: https://git-wip-us.apache.org/repos/asf?p=geode-examples.git;a=blob;f=micrometerMetrics/src/main/java/org/apache/geode_examples/micrometerMetrics/SimpleMetricsPublishingService.java;h=1880ce2bcc5110f9639875923bde4ca0a76e5463;hb=HEAD#l37

As long as we are not exposing the port to a public interface (i.e. public IP) I think this is good.